### PR TITLE
Fix return types: $this, static and self

### PR DESCRIPTION
**Pull request by distantnative**
Fixes https://github.com/getkirby/getkirby.com/issues/1223

### CODE REVIEW

**Comment by distantnative** [comment]
@lukasbestle If I see this correctly, the checks are failing because traits themselves can't really return `static`? Which I would kinda get but am surprised `self` worked before.

**Comment by lukasbestle** [comment]
Seems to be a bug in Psalm that only happens because of the `parent::update()` calls in these methods: https://github.com/vimeo/psalm/issues/5264

I think we should revert the changes in the two affected classes and add a TODO marker to them with a reference to the Psalm issue. Then we can already merge the rest.

File: src/Cms/Blueprint.php [hunk 1]
```diff
@@ -219,7 +219,7 @@ public static function extend($props): array
      * @param string $name
      * @param string|null $fallback
      * @param \Kirby\Cms\Model $model
-     * @return self
+     * @return static
```
Comment (lukasbestle) [inline src/Cms/Blueprint.php]:
```suggestion
     * @return static|null
```

File: src/Form/Fields.php [hunk 2]
```diff
@@ -23,7 +23,7 @@ class Fields extends Collection
      *
      * @param string $name
      * @param object|array $field
-     * @return self
+     * @return static
```
Comment (lukasbestle) [inline src/Form/Fields.php]: The return value is inherited from `Toolkit\Collection::__set()`

```suggestion
     * @return $this
```

**Review by lukasbestle — COMMENTED** [review]
Nice work, that turned out to be quite a long list of changes! 👍🙇

I only found two errors, the rest looks good.

**Review by lukasbestle — APPROVED** [review]

**Pull request merged**